aten/src/ATen/native/cpu/batch_norm_kernel.cpp

```diff
@@ -192,27 +189,44 @@ batch_norm_cpu_collect_stats_contiguous_impl(
   // parallel dim reduce on 'channel'
   at::parallel_for(0, n_channel, 1, [&](int64_t begin, int64_t end) {
     for (const auto c : c10::irange(begin, end)) {
-      // compute mean per input
-      accscalar_t sum = 0;
-      for (const auto n : c10::irange(n_batch)) {
-        for (const auto i : c10::irange(image_size)) {
-          auto offset = n * n_channel * image_size + c * image_size + i;
-          sum += input_data[offset];
+    // compute mean per input
+      scalar_t sum_val = 0;
+      Vec sum_val_vec = Vec(scalar_t(0));
```

Comment:
you can use a simple `vec::map_reduce` so save some effort: https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/cpu/vec/functional_base.h#L137